### DISCUSSION TITLE
Wrap right side of string in "%" for wildcard match in API rule

### DISCUSSION THREAD

**Discussion by towbeeWee** [pocketbase/pocketbase#4451] [Q&A] [answered]
Hello everyone, in my project I have a collection "pages" that lists pages of my project with a virtual path. Some virtual paths AND their subpaths should only be accessible by certain users (groups). With the API rules below user1 and user2 have access to the `/abc` and `/abc/def` paths. That is correct. However, user2 also has access to `/def/abc`. That is not correct.

**collection pages**
| name		| slug 		| path				|
|--			|--			|--					|
| page1	| /page1	| /abc				|
| page2	| /page2	| /abc/def		|
| page3	| /page3	| /def/abc		|

**collection permissions**
| path		| users 			|
|--			|--					|
| /abc		| user1, user2	|
| /def		| user1			|

**API Rules collection pages (read)**
```
@request.auth.id != "" && 
(path ?~ @collection.permissions.path && 
@request.auth.id ?~ @collection.permissions.users.id)
```

Is there a way to adjust the API rule so that the wildcard filter only includes the right side with %?
For example, something like this:
```
path ?~ @collection.permissions.path+"%"
```
Because this is obviously possible:
```
path ?~ "path_%"
```

I would be very happy about any help, even if I have to change the database model. Thanks in advance.

**Comment by ganigeorgiev** [comment, marked as answer]
Concatenation of fields is not supported at the moment. Specifying the wildcard for a dynamic field is also not supported (I'll consider it during the refactoring).

I'm not sure that I fully understand your use case but you can try to use a different delimiter for your root/starting path (eg. double slash `//` or similar).
An alternative could be to create a helper view collection with your desired column value dynamically calculated and join that in your API rule.

You could also replace `@request.auth.id ?~ @collection.permissions.users.id` with `@request.auth.id ?= @collection.permissions.users.id` for an exact match since you already target the individual id with `.users.id`.

**Reply by towbeeWee** [reply]
What I wanted to achieve is a virtual folder-based access system to specific pages or subfolders.
The double slash was already as backup in my mind :)
I found your suggestion with the helper view collection very elegant and it seems to work as desired with the first few attempts. I will continue working on it. Thank's for the Tipps.
